tests/aws/scenario/kinesis_firehose/test_kinesis_firehose.py

```diff
@@ -0,0 +1,192 @@
+        )
+        policy_firehose_kinesis = iam.Policy(
+            stack,
+            "FirehoseKinesisPolicy",
+            policy_name="firehose-kinesis-policy",
+            statements=[
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=[
+                        "kinesis:DescribeStream",
+                        "kinesis:GetShardIterator",
+                        "kinesis:GetRecords",
+                        "kinesis:ListShards",
+                    ],
+                    resources=[kinesis_stream.stream_arn],
+                ),
+            ],
+        )
+        role_firehose_kinesis.attach_inline_policy(policy_firehose_kinesis)
+
+        kinesis_stream_source_configuration = (
+            firehose.CfnDeliveryStream.KinesisStreamSourceConfigurationProperty(
+                kinesis_stream_arn=kinesis_stream.stream_arn,
+                role_arn=role_firehose_kinesis.role_arn,
+            )
+        )
+
+        # cloud watch logging group and stream for firehose s3 error logging
+        firehose_s3_log_group_name = "firehose-s3-log-group"
+        firehose_s3_log_stream_name = "firehose-s3-log-stream"
+        firehose_s3_log_group = logs.LogGroup(
+            stack,
+            "FirehoseLogGroup",
+            log_group_name=firehose_s3_log_group_name,
+            removal_policy=cdk.RemovalPolicy.DESTROY,  # required since default value is RETAIN
+        )
+        firehose_s3_log_group.add_stream(firehose_s3_log_stream_name)
+
+        # s3 access role for firehose
+        role_firehose_s3 = iam.Role(
+            stack,
+            "FirehoseS3Role",
+            role_name="firehose-s3-role",
+            assumed_by=iam.ServicePrincipal("firehose.amazonaws.com"),
+        )
+        policy_firehose_s3 = iam.Policy(
+            stack,
+            "FirehoseS3Policy",
+            policy_name="firehose-s3-policy",
+            statements=[
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=[
+                        "s3:AbortMultipartUpload",
+                        "s3:GetBucketLocation",
+                        "s3:GetObject",
+                        "s3:ListBucket",
+                        "s3:ListBucketMultipartUploads",
+                        "s3:PutObject",
+                    ],
+                    resources=[bucket.bucket_arn, f"{bucket.bucket_arn}/*"],
+                ),
+                iam.PolicyStatement(
+                    effect=iam.Effect.ALLOW,
+                    actions=["logs:PutLogEvents", "logs:CreateLogStream"],
+                    resources=[firehose_s3_log_group.log_group_arn],
+                ),
+            ],
+        )
+        role_firehose_s3.attach_inline_policy(policy_firehose_s3)
+
+        extended_s3_destination_configuration = firehose.CfnDeliveryStream.ExtendedS3DestinationConfigurationProperty(
```

Comment:
Idea: when this scenario is fleshed out, it would be cool to build a `Construct` for this resource so we don't have to deal with manual roles/policies and some of this destination configuration can be streamlined